File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Changes/WhatIsChangesApi.cs
using System;

using Raven.Client.Changes;
using Raven.Client.Document;

namespace Raven.Documentation.Samples.ClientApi.Changes
{
	public class WhatIsChangesApi
	{
		 private interface IFoo
		 {
			 #region changes_1
			 IDatabaseChanges Changes(string database = null);
			 #endregion
		 }

		public WhatIsChangesApi()
		{
			using (var store = new DocumentStore())
			{
				#region changes_2
				IDisposable subscription = store
					.Changes()
					.ForAllDocuments()
					.Subscribe(change => Console.WriteLine("{0} on document {1}", change.Type, change.Id));

				try
				{
					// application code here
				}
				finally
				{
					if (subscription != null)
						subscription.Dispose();
				}
				#endregion
			}
		}
	}
}

File: Documentation/6.1/Samples/java/src/test/java/net/ravendb/ClientApi/Changes/WhatIsChangesApi.java
package net.ravendb.ClientApi.Changes;

import net.ravendb.client.documents.DocumentStore;
import net.ravendb.client.documents.IDocumentStore;
import net.ravendb.client.documents.changes.IDatabaseChanges;
import net.ravendb.client.documents.changes.Observers;

public class WhatIsChangesApi {

    private interface IFoo {
        //region changes_1
        IDatabaseChanges changes();

        IDatabaseChanges changes(String database);
        //endregion
    }

    public WhatIsChangesApi() {
        try (IDocumentStore store = new DocumentStore()) {
            //region changes_2
            IDatabaseChanges subscription = store.changes();

            subscription.ensureConnectedNow();

            subscription.forAllDocuments().subscribe(Observers.create(change -> {
                System.out.println(change.getType() + " on document " + change.getId());
            }));

            try {
                // application code here
            } finally {
                if (subscription != null) {
                    subscription.close();
                }
            }
            //endregion
        }
    }
}
